lib/commands/change-environment.coffee

```diff
@@ -0,0 +1,75 @@
+module.exports = () ->
+  # ignore environments in these scopes
+  ignoredScopes = ["comment", "markup.raw.verbatim"]
+  te = atom.workspace.getActiveTextEditor()
+  cursorEnv = ({cursor: cursor, stack: []} for cursor in te.getCursorBufferPositions())
+  envRegex = /\\(begin|end)(?:\[[^\]]*\])?\{([^\}]*)\}/g
+  te.scan(envRegex, (matchObj) ->
+    match = matchObj.match
+    name = match[2]
+    isBegin = match[1] == "begin"
+    # get the outer point of the environment
+    envBorder = if isBegin then matchObj.range.start else matchObj.range.end
+    # ignore comments
+    descriptor = te.scopeDescriptorForBufferPosition(envBorder)
+    if (descriptor.getScopesArray().some (d) -> (ignoredScopes.some (s) -> d.startsWith(s)))
+      return
+    # retrieve information about the match object
+    endPoint = matchObj.range.end
+    column = endPoint.column - 1
+    range = [[endPoint.row, column - name.length], [endPoint.row, column]]
+
+    cursorEnv.forEach (envObj) ->
+      # if the environment is still before the cursor
+      if envObj["cursor"].isGreaterThan(envBorder)
```

Comment:
Small point but `envObj.cursor` is equivalent to `envObj["cursor"]`. I find the former more readable, but it really doesn't matter.